dexs/fastjpeg/index.ts

```diff
@@ -0,0 +1,71 @@
+import { parseEther } from "ethers";
+import { FetchOptions, SimpleAdapter } from "../../adapters/types";
+import { CHAIN } from "../../helpers/chains";
+
+const SwapCoinEvent = "event SwapCoin(address indexed sender, address indexed coin, uint256 amountA, uint256 amountB, uint256 volume, uint8 side)"
+const FastJpegFactory = '0x3BB7FBEeE877BD240de72A89AFd806BD3C1C8034';
+// Constants
+const UNDERGRADUATE_SUPPLY = parseEther('800000000'); // Example value, replace with actual
```

Comment:
is this supply fixed?

Reply:
Yeah, the contract supply is capped at 1 billion.
800m undergraduate for purchase on bonding curve before graduation
200m for bonding post graduation..